Obsidian.API/Commands/CommandModuleBase.cs

```diff
@@ -0,0 +1,34 @@
+﻿using Obsidian.API.Plugins;
+using System.Diagnostics;
+
+namespace Obsidian.API.Commands;
+public abstract class CommandModuleBase
+{
+    private CommandContext? commandContext;
+
+    [CommandContext]
+    public CommandContext CommandContext
+    {
+        get
+        {
+            if (commandContext == null)
+                throw new UnreachableException();//TODO empty command context maybe??
```

Comment:
You can't really do anything without a context so I think just letting it throw is valid.